tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/graph_executions/graph_executions_container_test.ts

```diff
@@ -64,14 +68,93 @@ describe('Graph Executions Container', () => {
     store = TestBed.get(Store);
   });
 
-  it('renders number of graph executions', () => {
+  it('does not render execs viewport if # execs = 0', fakeAsync(() => {
+    const fixture = TestBed.createComponent(GraphExecutionsContainer);
+    store.overrideSelector(getNumGraphExecutions, 0);
+    fixture.autoDetectChanges();
```

Comment:
Yes, the special test patterns you see here (including `autoDetectChanges()`, `tick()` and `fakeAsync()`) are used because of the virtual scrolling used by the component.

- `autoDetectChanges()` seems to be required by the cdk scrolling. I couldn't get the test to pass by using `refreshState()` and `detectChanges()`.
- As for `tick(500)`, I found it can be replaced with simply `tick()`. Done.
- `fakeAsync()` is required for `tick()`. See https://angular.io/api/core/testing/tick.